### ISSUE TITLE
Only allow certain remote references to be resolved

### ISSUE THREAD

**Issue by nulltoken** [stoplightio/spectral#436]
**User story.**
As a API manager, I can add a rule that would restrict the list of automatically resolved http external references to a known subset, so that I can eventually ensure that only vetted external references are used.

**Describe the solution you'd like**
A way to configure spectral so that a list of complete urls (or regular expressions) describing the http remote references that are allowed to be resolved.

This could either be a new core rule or, maybe simpler from an implementation standpoint, an option to the http resolver.

**Additional context**
- From an Enterprisey persective, one may need to restrict the domains from which remote refs are allowed to be resolved from.
- Those remote references may actually point to versioned schemas, and one would want to only allow supported/stable versions to be used.

**Comment by nulltoken** [comment]
I believe the ´unresolved’ from @p0lip’s #473 should make that easier to implement

**Comment by P0lip** [comment]
@nulltoken 
Yeah, you are right.
Creating a rule to lint $refs is totally feasible as of 4.1.0.
An example rule disallowing `foo.com` origin could be written as follows
```yaml
 disallow-refs:
    message: $ref must not point at disallowed remotes
    given: $..$ref
    type: validation
    recommended: true
    resolved: false
    then:
      function: pattern
      functionOptions:
        notMatch: "\\/\\/foo\\.com\\/"
```

**Comment by P0lip** [comment]
@nulltoken did the above rule do the trick?

**Comment by P0lip** [comment]
Alright, thanks for letting me know!
I'll close the issue.
If you have any further concerns, feel free to re-open.
